velox/exec/fuzzer/CacheFuzzer.cpp

```diff
@@ -0,0 +1,364 @@
+  if (fileIds_.empty()) {
+    for (auto i = 0; i < FLAGS_num_source_files; ++i) {
+      const auto fileName =
+          fmt::format("{}/file_{}", sourceDataDir_->getPath(), i);
+      const size_t fileSize = boost::random::uniform_int_distribution<int64_t>(
+          FLAGS_min_source_file_bytes, FLAGS_max_source_file_bytes)(rng_);
+      auto writeFile = fs_->openFileForWrite(fileName);
+      size_t writtenSize = 0;
+      int32_t offset = 0;
+      while (writtenSize < fileSize) {
+        const size_t chunkSize = std::min(
+            fileSize - writtenSize,
+            size_t(4 << 20)); // Write in chunks of 4MB
+        auto buffer = folly::IOBuf::create(chunkSize);
+        buffer->append(chunkSize);
+        // Fill buffer with data.
+        std::generate_n(
+            buffer->writableData(), chunkSize, [&offset]() -> uint8_t {
+              return static_cast<uint8_t>(offset++ % 256);
+            });
+        writeFile->append(std::move(buffer));
+        writtenSize += chunkSize;
+      }
+      writeFile->close();
+
+      fileNames_.emplace_back(fileName);
+      fileIds_.emplace_back(fileIds(), fileName);
+      fileSizes_.emplace_back(fileSize);
+    }
+  }
+}
+
+void CacheFuzzer::initializeCache() {
+  // We have up to 20 threads and 16 threads are used for reading so
+  // there are some threads left over for SSD background write.
+  executor_ = std::make_unique<folly::IOThreadPoolExecutor>(20);
+
+  std::unique_ptr<SsdCache> ssdCache;
+  if (FLAGS_ssd_cache_bytes > 0) {
+    SsdCache::Config config(
+        fmt::format("{}/cache", sourceDataDir_->getPath()),
+        FLAGS_ssd_cache_bytes,
+        FLAGS_num_ssd_cache_shards,
+        executor_.get(),
+        FLAGS_ssd_checkpoint_interval_bytes,
+        false,
+        FLAGS_enable_checksum,
+        FLAGS_enable_checksum_read_verification);
+    ssdCache = std::make_unique<SsdCache>(config);
+  }
+
+  memory::MemoryManagerOptions options;
+  options.useMmapAllocator = true;
+  options.allocatorCapacity = FLAGS_memory_cache_bytes;
+  options.arbitratorCapacity = FLAGS_memory_cache_bytes;
+  options.arbitratorReservedCapacity = 0;
+  options.trackDefaultUsage = true;
+  memoryManager_ = std::make_unique<memory::MemoryManager>(options);
+
+  // TODO: Test different ssd write behaviors with AsyncDataCache::Options.
+  cache_ = AsyncDataCache::create(
+      dynamic_cast<memory::MmapAllocator*>(memoryManager_->allocator()),
+      std::move(ssdCache),
+      {});
+}
+
+void CacheFuzzer::initializeInputs() {
+  const auto readOptions = io::ReaderOptions(pool_.get());
+  auto tracker = std::make_shared<ScanTracker>(
+      "testTracker", nullptr, 256 << 10 /*256KB*/);
+  auto ioStats = std::make_shared<IoStatistics>();
+  inputs_.reserve(FLAGS_num_source_files);
+  for (auto i = 0; i < FLAGS_num_source_files; ++i) {
+    // Initialize buffered input.
+    auto readFile = fs_->openFileForRead(fileNames_[i]);
+    groupIds_.emplace_back(
```

Comment:
Do we need the file group support? Thanks!

Reply:
Probably not. Let me remove it.